src/NLog/LayoutRenderers/DateLayoutRenderer.cs

```diff
@@ -74,21 +75,69 @@ public DateLayoutRenderer()
         [DefaultValue(false)]
         public bool UniversalTime { get; set; }
 
+        private static char[] lowTimeResolutionChars = new char[] { 'Y', 'y', 'M', 'D', 'd', 'H', 'h' };
+        private System.Collections.Generic.KeyValuePair<DateTime, string> cachedUtcTime = new System.Collections.Generic.KeyValuePair<DateTime, string>();
```

Comment:
I think the  key is the datetime to be used, and the value the format? I need a small explanation :)